doc/adr/0016- end-to-end testing-strategy.rst

```diff
@@ -0,0 +1,199 @@
+  will often be prioritised.
+* This approach will will improve our high-level perspective of each componenent and help guide
+  UX improvements.
+* Node team are not pressured to focus on plutus scenarios, they retain control of their priorities.
+
+Cons of building and maintaining our own test framework
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* ``cardano-node-tests`` is well established and already has useful features such as: running tests in
+  different eras, transitioning between eras, reporting, and measuring deviations in script cost.
+* We are proposing to add another means of spinning up a local testnet when there are already
+  multiple others (see notes section below).
+* No major work needed to continue using ``cardano-node-tests``. Node test team will continue to
+  upgrade/maintain it regardless.
+* Some coverage of plutus tests already exist and can continue to be added in
+  ``cardano-node-tests`` by the node qa and plutus teams.
+* Going forwards, many plutus tests will be mostly a copy/paste job. For example, the existing
+  SECP256k1 tests build and submit a single transaction using a basic minting policy and a redeemer
+  consisting of inputs to the verify function. This approach will be similar to future tests of
+  builtin functions.
+* Plutus team will still be required to support the node test team with defining and reviewing a
+  subset of plutus tests in ``cardano-node-tests``.
+* Node test team may grow, less delays in getting plutus tests implementd by a Python developer.
+* ``cardano-cli`` is a wrapper for ``cardano-api`` (soon to be ``cardano-ledger-api`` and
+  ``cardano-node-client``), so that package already getting test coverage at the end-to-end level.
+
+Additional Considerations
+-------------------------
+* Business stakeholders will want to see test results so think about producing and storing a report.
+  It would be nice open source this along with the tests.
+* At first tests will be run on private testnet but we must consider how these tests can also be run
+  on a public testnet. For example, initial wallet balances and utxos will need to be handled
+  dynamically because we'd only have control over these in the private testnet.
+* Seeing as ``cardano-ledger-api`` and ``cardano-node-client`` are still in early stages of production
+  it would make sense not to block creation of this test framework. We can begin using ``cardano-api``
+  and switch over when ready.
+* End-to-end tests can be slow to execute and as the suite grows we may want to run a subset at more
+  frequent intervals. For example, we run tests for the latest Plutus version nightly but older
+  tests/versions are run weekly, or for tags/release only.
+
+Alternatives
+------------
+
+Other than choosing to remain using ``cardano-node-test``, we could look at the approach teams such as
+Djed or Hydra are using to see if we can share infrastructure. I don't think either are using
+``cardano-testnet``.
```

Comment:
I don't think there's much to share. Djed or Hydra basically have their own bash scripts to spin up a local testnet. There is no argument to use this instead of `cardano-testnet`, which officially supported by the node team.